### PR TITLE
[Fleet] Added logs-elastic_agent* read privileges to kibana_system

### PR DESCRIPTION
**Pull request by juliaElastic**
Required change for Fleet telemetry to read from elastic agent log indices.
https://github.com/elastic/kibana/pull/146107

### CODE REVIEW

**Comment by juliaElastic** [comment]
I see some ml tests failing, which is unrelated to my changes. Any idea how to fix this? I just did a merge from main as well.
```
15:53:26 * What went wrong:
15:53:26 Execution failed for task ':x-pack:plugin:ml:test'.
15:53:26 > There were failing tests. See the report at: file:///dev/shm/elastic+elasticsearch+pull-request+part-3-fips/x-pack/plugin/ml/build/reports/tests/test/index.html
15:53:26
```

**Comment by juliaElastic** [comment]
Jenkins, test this please

**Comment by droberts195** [comment]
Jenkins run elasticsearch-ci/part-3-fips

**Comment by ywangd** [comment]
Ping @elastic/kibana-security

File: x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/security/authz/store/ReservedRolesStore.java
Comment (ywangd) [inline x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/security/authz/store/ReservedRolesStore.java:723]: I am mostly curious: Does Fleet actually need `kibana_system` to read cross cluster for these indices?

Comment (juliaElastic) [reply x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/security/authz/store/ReservedRolesStore.java:723]: I don't think it is needed right now, I mostly copied the other roles. Should I remove it?

Comment (juliaElastic) [reply x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/security/authz/store/ReservedRolesStore.java]: removed

**Comment by juliaElastic** [comment]
@ywangd are you ready to approve?

File: x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/security/authz/store/ReservedRolesStore.java
Comment (juliaElastic) [reply x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/security/authz/store/ReservedRolesStore.java:723]: Not yet, I'll wait for his feedback.

Comment (juliaElastic) [reply x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/security/authz/store/ReservedRolesStore.java:723]: See https://github.com/elastic/kibana/pull/145353#issuecomment-1323529760

Comment (juliaElastic) [reply x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/security/authz/store/ReservedRolesStore.java:723]: Is there anything else you need to approve?

Comment (azasypkin) [reply x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/security/authz/store/ReservedRolesStore.java:723]: Nope, everything looks good from Kibana perspective, feel free to merge as soon as ES team approves the change.

Comment (juliaElastic) [reply x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/security/authz/store/ReservedRolesStore.java:723]: @elastic/elasticsearch-team Hey, could someone review and approve?

Comment (slobodanadamovic) [reply x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/security/authz/store/ReservedRolesStore.java:723]: @juliaElastic 
We try to advise people not to use `@elastic/elasticsearch-team` handle, since it pings the whole Elasticsearch team (116 people).

I see that you already notified the correct team (`Team:Security`) and that Yang (who is on PTO) did the initial review. Everyone from Security team receives notifications for every comment, so you can either simply comment on this PR or ping `@elastic/es-security` if there is an urgent need for someone to have a look. 

From my perspective, the changes look good! 🚀  If you need this to be backported to `8.6` branch you can simply apply the `auto-backport-and-merge` label (before merging this PR) and it will automatically open a new PR and merge it (otherwise you would have to do it manually).

Another small suggestion would be to update the description of this PR (as it serves for future references) and remove contributor template bullet points and the link to a private GitHub repo.

**Review by slobodanadamovic — APPROVED** [review]

File: x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/security/authz/store/ReservedRolesStore.java
Comment (juliaElastic) [reply x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/security/authz/store/ReservedRolesStore.java:723]: Thank you for your suggestions! I wasn't sure which team to ping, as there was no team assigned as pr reviewer. I'll keep this in mind.

Comment (slobodanadamovic) [reply x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/security/authz/store/ReservedRolesStore.java:723]: I've made the same mistakes myself.  It's not something you could have known, so no worries. :)

**Pull request merged**
